lib/ansible/modules/cloud/amazon/rds_instance_facts.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/python
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: rds_instance_facts
+version_added: "2.4"
+short_description: obtain facts about one or more RDS instances
+description:
+  - obtain facts about one or more RDS instances
+options:
+  name:
+    description:
+      - Name of an RDS instance.
+    required: false
+    aliases:
+      - name
+  filters:
+    description:
+      - A filter that specifies one or more DB instances to describe.
+        See U(https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DescribeDBInstances.html)
+requirements:
+    - "python >= 2.6"
+    - "boto3"
+author:
+    - "Will Thames (@willthames)"
+    - "Michael De La Rue (@mikedlr)"
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Get facts about an instance
+- rds_instance_facts:
+    name: new-database
+  register: new_database_facts
+
+# Get all RDS instances
+- rds_instance_facts:
+'''
+
+RETURN = '''
+instances:
+    description: zero or more instances that match the (optional) parameters
+    type: list
+    returned: always
+    sample:
+       "instances": [
+           {
+               "availability_zone": "ap-southeast-2c",
+               "backup_retention": 10,
+               "create_time": "2017-02-20T06:13:05.724000+00:00",
+               "db_engine": "postgres",
+               "endpoint": "helloworld-rds-master.cyaju3p4tyqv.ap-southeast-2.rds.amazonaws.com",
+               "id": "helloworld-rds-master",
+               "instance_type": "db.t2.micro",
+               "maintenance_window": "sun:15:40-sun:16:10",
+               "multi_zone": false,
+               "port": 5432,
+               "replication_source": null,
+               "size": 5,
+               "status": "available",
+               "username": "hello",
+               "vpc_security_groups": "sg-abcd1234"
+           },
+       ]
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import get_aws_connection_info, boto3_conn, ansible_dict_to_boto3_filter_list
+from ansible.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+from ansible.module_utils.aws.rds import RDSDBInstance
+
+try:
+    import botocore
+except BaseException:
```

Comment:
It's inherited code (from other people via you I guess.. ) I don't know why not ImportError and will change it if you don't know.  At least that way we would find out if it was important ;-)